packages/core/src/internal/proc.js

```diff
@@ -534,73 +534,63 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
     // Fork effects are non cancellables
   }
 
-  function runJoinEffect(t, cb) {
-    if (t.isRunning()) {
+  function runJoinEffect(taskOrTasks, cb) {
+    if (is.array(taskOrTasks)) {
+      if (taskOrTasks.length === 0) {
+        cb([])
+        return
+      }
+
+      const childCallbacks = createAllStyleChildCallbacks(taskOrTasks, cb)
```

Comment:
`createAllStyleChildCallbacks` won't work on empty array or object. should we add such a check there?

Reply:
Agree.  Though `createAllStyleChildCallbacks` is internal only, it makes sense to add such a check.